### ISSUE TITLE
Flux unable to pull git repo from Github Enterprise

### ISSUE THREAD

**Issue by RochesterinNYC** [weaveworks/flux#889]
I deployed Flux and all its associated resources to a Kubernetes cluster in the `company-system` namespace. 

Locally, I then run the following to forward local requests to port 10080 to port 3030 of the flux deployment:
```
flux_pod=$(kubectl -n company-system get pod -l name=flux -o name | awk -F / '{ print $2; }')
kubectl -n company-system port-forward "$flux_pod" 10080:3030
``` 

Locally, when I run `fluxctl` commands, I get the following error message:

```
$ fluxctl list-images
== Error ==

Problem cloning your git repository

There was a problem cloning your git repository,

    git@ghe.company.net:jameswen/kubernetes-monorepo-prototype

This may be because you have not supplied a valid deploy key, or
because the repository has been moved, deleted, or never existed.

Please check that there is a repository at the address above, and that
there is a deploy key with write permissions to the repository. In
GitHub, you can do this via the settings for the repository, and
cross-check with the fingerprint given by

    fluxctl identity
```

Before deploying Flux, I had generated the SSH key locally, added the public key as a deploy key with write access to the jameswen/kubernetes-monorepo-prototype GHE repo. I verified that the key fingerprint shown by `fluxctl identity -l` matches the fingerprint of this deploy key on GHE.

Additionally, I kubectl exec-ed into the running flux pod/container, started up the ssh-agent (or attached to the already running one), added the SSH private key found in the container at the path I specified (`/etc/fluxd/ssh/identity`) to the ssh-agent, and then tried to clone the jameswen/kubernetes-monorepo-prototype GHE repo and it successfully cloned (which makes it seem like it is not a DNS resolution or network connectivity/firewall issue).

# Extra Information

All deployed Flux resources:
```
---
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: flux
  namespace: company-system
spec:
  replicas: 1
  strategy:
    type: Recreate
  template:
    metadata:
      labels:
        name: flux
    spec:
      # The serviceAccount is only needed for Kubernetes with RBAC.
      # You will also need to create the resources in flux-account.yaml.
      serviceAccount: flux
      volumes:
      - name: git-key
        secret:
          secretName: flux-git-deploy
      containers:
      - name: flux
        # There are no ":latest" images for flux. Find the most recent
        # release or image version at https://quay.io/weaveworks/flux
        # and replace the tag here.
        image: quay.io/weaveworks/flux:1.1.0
        imagePullPolicy: IfNotPresent
        ports:
        - containerPort: 3030 # informational
        volumeMounts:
        - name: git-key
          mountPath: /etc/fluxd/ssh
        args:

        # if you deployed memcached in a different namespace to flux,
        # or with a different service name, you can supply these
        # following two arguments to tell fluxd how to connect to it.
        # - --memcached-hostname=memcached.default.svc.cluster.local
        # - --memcached-service=memcached

        # replace (at least) the following URL
        - --git-url=git@ghe.company.net:jameswen/kubernetes-monorepo-prototype
        - --git-branch=master
---
# The service account, cluster roles, and cluster role binding are
# only needed for Kubernetes with role-based access control (RBAC).
apiVersion: v1
kind: ServiceAccount
metadata:
  labels:
    name: flux
  name: flux
  namespace: company-system
---
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRole
metadata:
  labels:
    name: flux
  name: flux
  namespace: company-system
rules:
  - apiGroups: ['*']
    resources: ['*']
    verbs: ['*']
  - nonResourceURLs: ['*']
    verbs: ['*']
---
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRoleBinding
metadata:
  labels:
    name: flux
  name: flux
  namespace: company-system
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: flux
subjects:
  - kind: ServiceAccount
    name: flux
    namespace: company-system
---
apiVersion: v1
kind: Secret
metadata:
  name: flux-git-deploy
  namespace: company-system
data:
  identity: <redacted base64 value of private SSH key>
---
# This defines a Kubernetes service for the Flux daemon, so you can
# access it with `fluxctl`. It defines a NodePort, meaning Kubernetes
# will assign a port to the service, on which it will be available on
# each node (host in the cluster). You can find the assigned port by
# doing something like
#
#    kubectl get service flux --template '{{ index .spec.ports 0 "nodePort" }}'
#
apiVersion: v1
kind: Service
metadata:
  name: flux
  namespace: company-system
spec:
  type: NodePort
  ports:
    - port: 80
      targetPort: 3030
  selector:
    name: flux
---
# You can optionally deploy memcache, for the Flux daemon to cache
# container image metadata.
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: memcached
  namespace: company-system
spec:
  replicas: 1
  template:
    metadata:
      labels:
        name: memcached
    spec:
      containers:
      - name: memcached
        image: memcached:1.4.25
        imagePullPolicy: IfNotPresent
        args:
        - -m 64    # Maximum memory to use, in megabytes. 64MB is default.
        - -p 11211    # Default port, but being explicit is nice.
        - -vv    # This gets us to the level of request logs.
        ports:
        - name: clients
          containerPort: 11211
---
apiVersion: v1
kind: Service
metadata:
  name: memcached
  namespace: company-system
spec:
  # The memcache client uses DNS to get a list of memcached servers and then
  # uses a consistent hash of the key to determine which server to pick.
  clusterIP: None
  ports:
    - name: memcached
      port: 11211
  selector:
    name: memcached
```

**Comment by squaremo** [comment]
Off the top of my head, this may be related to https://github.com/weaveworks/flux/issues/724. Short version: we pre-install an SSH known_hosts file, and git (over SSH) will refuse to clone a repo from a host not in that file. I've posted a possible workaround in that issue.

**Comment by RochesterinNYC** [comment]
The recommended solution by @squaremo in #724 fixed this issue for me (using version 1.2.0 and mounting a known_hosts configmap/file to `/root/.ssh`.
